File: catalog-service/src/main/java/org/nsrfarms/entity/Category.java
package org.nsrfarms.entity;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.Data;
import lombok.ToString;

@Data
@ToString
@Entity
public class Category {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;
	private String name;

	private String description;

	@OneToMany(fetch = FetchType.LAZY, mappedBy = "category" )
	private List<Product> products;
	
	@JsonIgnore
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "catalog_id")
	private Catalog catalog;
}

File: typescript-demo/Category.ts
export class Category{
    constructor(private _name:string,private _desc:string){

    }
    public get name(): string {
      return this._name;
    }
    public set name(value: string) {
      this._name = value;
    }
    public get desc(): string {
      return this._desc;
    }
    public set desc(value: string) {
      this._desc = value;
    }

    
    
}
